demo-app-benchmark-test/src/main/java/cash/z/ecc/android/sdk/demoapp/benchmark/StartupBenchmark.kt

```diff
@@ -0,0 +1,111 @@
+package cash.z.ecc.android.sdk.demoapp.benchmark
+
+import androidx.benchmark.macro.CompilationMode
+import androidx.benchmark.macro.ExperimentalMetricApi
+import androidx.benchmark.macro.MacrobenchmarkScope
+import androidx.benchmark.macro.StartupMode
+import androidx.benchmark.macro.StartupTimingMetric
+import androidx.benchmark.macro.TraceSectionMetric
+import androidx.benchmark.macro.junit4.MacrobenchmarkRule
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Until
+import org.junit.Rule
+import org.junit.Test
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * Purpose of this class is to provide a basic startup measurements, and captured system traces for investigating the
+ * app's performance. It navigates to the device's home screen, and launches the default activity.
+ *
+ * Run this benchmark from Android Studio only against the Benchmark build type set in all modules.
+ *
+ * We ideally run this against a physical device with Android SDK level 29, at least, as profiling is provided by this
+ * version and later on.
+ */
+class StartupBenchmark {
```

Comment:
Should we have an assertion that checks that the screen is turned on before the benchmark runs?

This won’t be a problem for a virtual device, but on a physical device it might be helpful since the time between starting a build and the tests launching might be 30+ seconds.

We have something in our UI instrumentation tests, but it might be helpful here too.

Reply:
Yes, good idea. Added.